lib/logstash/outputs/mongodb.rb

```diff
@@ -62,11 +53,11 @@ def receive(event)
       if @generateId
         document['_id'] = BSON::ObjectId.new(nil, event["@timestamp"])
       end
-      @db.collection(event.sprintf(@collection)).insert(document)
+      @db[event.sprintf(@collection)].insert_one(document)
     rescue => e
       @logger.warn("Failed to send event to MongoDB", :event => event, :exception => e,
                    :backtrace => e.backtrace)
-      if e.error_code == 11000
+      if e.message =~ /^11000/
```

Comment:
This needs to be /^E11000/

Here's a snippet of the error: `:exception=>#<Mongo::Error::OperationFailure: E11000 duplicate key error...`